docs/conf.py

```diff
@@ -164,3 +162,29 @@
 # so a file named "default.css" will overwrite the builtin "default.css".
 html_static_path = ["_static"]
 html_css_files = ["custom.css"]
+
+# --- custom gallery of websites -----------------------------------------------
+from pathlib import Path
+import json
+from shutil import copy
+from playwright.sync_api import sync_playwright
+
+
+gallery_item = (Path(__file__).parent / "_templates/gallery_item.rst").read_text()
+gallery = json.loads((Path(__file__).parent / "_templates/gallery.json").read_text())
+src = Path(__file__).parent / "_templates/gallery.rst"
+dst = Path(__file__).parent / "user_guide/gallery.rst"
+copy(src, dst)
+
+
+with dst.open("a") as f, sync_playwright() as p:
+
+    browser = p.chromium.launch()
+    page = browser.new_page()
+
+    for item in gallery:
+        item["id"] = item["name"].lower().replace(" ", "_")
+        page.goto(item["website"])
+        page.screenshot(path=f"_static/gallery/{item['id']}.png")
+
+        f.write(gallery_item.format(**item))
```

Comment:
I'd recommend doing a check for whether a local file already exists, and only run all of this playwright code if it _doesn't_. That way people don't need to be running playwright every time the docs are built, which I think could be quite slow. It'd only happen when the person wanted to build the gallery from scratch.

Another option might be to use a flag that only build the gallery if it's on READTHEDOCS or something. I'm trying to think of how we could reduce the amount of extra build time this would add for development

Reply:
I added the check on the files (so if you build it locally they will be downloaded only once). Also all the plywright logic is encapsulated in a "try ... except" so timeout error will simply use the 404 screenshot that I found on the web (the exact search was "free 404 page")